### ISSUE TITLE
중간고사 이론 시험 4번 모범 답안

### ISSUE THREAD

**Issue by unizard** [sejongresearch/2020.Spring.AI#10]
- 문제
  - 선형 분류기 설계 시 이진분류기와 다중분류기의 차이점을 설명하시오.
- 정답 포인트
  - sigmoid와 softmax 언급
  - 데이터 정규화와 전체의 합이 1 언급

**Comment by unizard** [comment]
// 18011817
이진분류기를 설계할 때, 가설에 sigmoid 함수를 사용함으로써 예측값을 0과 1 사이로 제한한다. 그 결과 0.5보다 크면 1, 0.5보다 작으면 0으로 분류할 수 있게된다. 그러나 기존 sigmoid 함수를 다중분류기에 적용하였을 때, 여러 값에 1이 나와 정확한 분류가 이루어지지 못한다는 단점이 발생한다. 따라서 다중분류기는 여전히 모든 값을 0~1 사이면서 전체 합이 1이 되도록 확률 정규화를 시키는 softmax 함수를 사용하여 하나의 class에만 1이 발생하여 분류할 수 있게 된다

**Comment by unizard** [comment]
// 18011784
- 이진 분류기 - 가설함수로 sigmoid를 사용함. cost function으로는 (-y*log(Hypothesis) - (1-y) * log(1-Hypothesis)).mean를 사용
- 다중 분류기 - 가설함수로 softmax를 사용함. softmax로 나온 값을 argmax를 통해 onehot encoding을 진행하며, 그 후 cross entropy라는 cost function을 사용함.

**Comment by unizard** [comment]
// 19011484

이진 분류기는 Hypothesis에 Sigmoid 함수를 씌우고, 이진 크로스 엔트로피 오차를 사용한다. 그렇기에 분류기의 출력이 0과 1 사이의 값을 가지며, 이때, 적절하게 올림과 내림을 사용하여 0,1로 분류한 후, 각각을 두 개의 클래스와 매치한다.

다중분류기는 Hypothesis에 Softmax 함수를 씌우고 크로스 엔트로피 오차를 사용한다.
출력은 각 클래스에 대한 0~1 사이의 값이 나오며, 출력의 모든 값을 더하면 1이 된다.
그렇기에 출력 중 가장 큰 값을 이용하여, 해당 클래스일 확률이 가장 높다고 볼 수 있다.
